### PR TITLE
Small clean-up + add new tests

### PR DESCRIPTION
**Pull request by aozherelyeva**
## Motivation and Context
- Add tests for the system message retention when compressing history ([KG-311](https://youtrack.jetbrains.com/issue/KG-311)) 
- Add tests for the empty system message with Anthropic model-based agents ([KG-317](https://youtrack.jetbrains.com/issue/KG-317))
- Update eventHandler methods from deprecated ones in the tests.

## Breaking Changes

None.

---

#### Type of the changes
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation update
- [x] Tests improvement
- [x] Refactoring

#### Checklist
- [x] The pull request has a description of the proposed change
- [x] I read the [Contributing Guidelines](https://github.com/JetBrains/koog/blob/main/CONTRIBUTING.md) before opening the pull request
- [x] The pull request uses **`develop`** as the base branch
- [x] Tests for the changes have been added
- [x] All new and existing tests passed

##### Additional steps for pull requests adding a new feature
- [x] An issue describing the proposed change exists
- [x] The pull request includes a link to the issue
- [ ] The change was discussed and approved in the issue
- [ ] Docs have been added / updated

### CODE REVIEW

File: integration-tests/src/jvmTest/kotlin/ai/koog/integration/tests/agent/AIAgentIntegrationTest.kt [hunk 1]
```diff
@@ -1188,4 +1240,84 @@ class AIAgentIntegrationTest {
             )
         }
     }
+
+    @ParameterizedTest
+    @MethodSource("historyCompressionStrategies")
+    fun integration_AIAgentHistoryCompression(strategy: HistoryCompressionStrategy, strategyName: String) =
```
Comment (antoniibelyshev) [inline integration-tests/src/jvmTest/kotlin/ai/koog/integration/tests/agent/AIAgentIntegrationTest.kt:1246]: There are more complications with history compression when tools are called. Could you please add a test for that case? Also, using a slightly longer conversation could be good. Probably, use the calculator agent from the examples, and compress the history in the end?

Comment (zarechneva-a) [reply integration-tests/src/jvmTest/kotlin/ai/koog/integration/tests/agent/AIAgentIntegrationTest.kt:1246]: Sounds like a plan, thank you! Will add this as an improvement in the next PR
